databases/week1/homework.sql

```diff
@@ -0,0 +1,79 @@
+-- Find out how many tasks are in the task table
+SELECT
+  count(*) as totaltasks
+FROM
+  task;
+--Find out how many tasks in the task table do not have a valid due date
+SELECT
+  COUNT(*) as notvalidduedate
+FROM
+  task
+WHERE
+  due_date IS NULL;
+--Find all the tasks that are marked as done
+SELECT
+  *
+from
+  task
+WHERE
+  status_id = 3;
+--Find all the tasks that are not marked as done
+SELECT
+  *
+from
+  task
+WHERE
+  status_id != 3;
+--Get all the tasks, sorted with the most recently created first
+SELECT
+  title,
+  created
+from
+  task
+ORDER BY
+  created DESC;
+--Get the single most recently created task
+SELECT
+  *
+FROM
+  task
+ORDER BY
+  created DESC
+LIMIT
+  1;
+--Get the title and due date of all tasks where the title or description contains database
+SELECT
+  title,
+  due_date
+from
+  task
+WHERE
+  title LIKE "%database%"
+  or description LIKE "%database%";
+--Get the title and status (as text) of all tasks.
+SELECT
+  task.title,
+  status.name AS text
+FROM
+  task
+  INNER JOIN status ON task.status_id = status.id;
```

Comment:
Why you are using the JOIN clause here? No difference between JOIN & INNER JOIN. So try to use one of them for good practice.
https://blog.sqlauthority.com/2013/07/16/sql-difference-between-inner-join-and-join/